cloud/services/loadbalancers.go

```diff
@@ -132,12 +137,21 @@ func AddNodeToNB(
 	if machineScope.LinodeCluster.Spec.Network.LoadBalancerPort != 0 {
 		lbPort = machineScope.LinodeCluster.Spec.Network.LoadBalancerPort
 	}
+
 	if machineScope.LinodeCluster.Spec.Network.NodeBalancerConfigID == nil {
 		err := errors.New("nil NodeBalancer Config ID")
 		logger.Error(err, "config ID for NodeBalancer is nil")
 
 		return err
 	}
+
+	if machineScope.LinodeCluster.Spec.Network.NodeBalancerID == nil {
+		err := errors.New("nil NodeBalancer ID")
+		logger.Error(err, "NodeBalancer ID is nil")
+
+		return err
+	}
```

Comment:
Similar to https://github.com/linode/cluster-api-provider-linode/pull/196/files#r1533129717 -- is `NodeBalancerID` optional? See https://github.com/linode/cluster-api-provider-linode/blob/main/api/v1alpha1/linodecluster_types.go#L108-L110. If not, we should validate in the API type, and not here.

Reply:
I'll revert it. Didn't realize it was optional :)